client/beefy/src/worker.rs

```diff
@@ -490,16 +495,19 @@ where
 					// worker won't vote until it witnesses a session change.
 					// Once we'll implement 'initial sync' (catch-up), the worker will be able to
 					// start voting right away.
-					self.handle_finality_notification(notif);
-					future::ready(false)
+					self.handle_finality_notification(&notif);
+					break
 				} else {
 					trace!(target: "beefy", "🥩 Finality notification: {:?}", notif);
 					debug!(target: "beefy", "🥩 Waiting for BEEFY pallet to become available...");
-					future::ready(true)
 				}
-			})
-			.for_each(|_| future::ready(()))
-			.await;
+				},
+				_ = gossip_engine.fuse() => {
```

Comment:
```suggestion
				_ = gossip_engine => {
```

Otherwise the fuse is useless.